### PR TITLE
Enable FUSE and NTFS support [ESD-1229]

### PR DESCRIPTION
**Pull request by silverjam**
## Design notes
Adds NTFS as an alternative to F2FS, and behaves identically.  FAT is expected to be the default file system for sd/usb logging, any transition from FAT (or unconfigured) to NTFS/F2FS, or between NTFS<->F2FS will re-provision the drive.  Moving from F2FS/NTFS back to FAT doesn't nothing to avoid any possibility of accidentally wiping customer data in the default configuration.

## Test
Very basic bench testing that NTFS works as a filesystem, need to test migration process more thoroughly, initial testing showed that the NTFS format process was very slow... so something to consider. @swiftnav-sarika to do more extensive testing soon.

### CODE REVIEW

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.2.0-develop-2019040519-3-g085f0859

These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.

### job status
+ [live-roof-650-townsend runs](https://gnss-analysis.swiftnav.com/jobs/id=989506)
+ [live-roof-650-townsend-dropouts-zero-baseline runs](https://gnss-analysis.swiftnav.com/jobs/id=989507)

### gnss-analysis results
At least one run must complete for these links to have data.

#### passfail
+ [live-roof-650-townsend](https://gnss-analysis.swiftnav.com/metrics_preset=pass_fail&scenario=live-roof-650-townsend&summary_type=q50&groupby_key=firmware&firmware_versions=v2.2.0-develop-2019040519-3-g085f0859&build_type=buildroot_pr&display_type=table)
+ [live-roof-650-townsend-dropouts-zero-baseline](https://gnss-analysis.swiftnav.com/metrics_preset=pass_fail&scenario=live-roof-650-townsend-dropouts-zero-baseline&summary_type=q50&groupby_key=firmware&firmware_versions=v2.2.0-develop-2019040519-3-g085f0859&build_type=buildroot_pr&display_type=table)

#### detailed
+ [live-roof-650-townsend](https://gnss-analysis.swiftnav.com/metrics_preset=detailed&scenario=live-roof-650-townsend&summary_type=q50&groupby_key=firmware&firmware_versions=v2.2.0-develop-2019040519-3-g085f0859&build_type=buildroot_pr&display_type=table)
+ [live-roof-650-townsend-dropouts-zero-baseline](https://gnss-analysis.swiftnav.com/metrics_preset=detailed&scenario=live-roof-650-townsend-dropouts-zero-baseline&summary_type=q50&groupby_key=firmware&firmware_versions=v2.2.0-develop-2019040519-3-g085f0859&build_type=buildroot_pr&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v2.2.0-develop-2019040519-3-g085f0859
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.2.0-develop-2019040519-3-g085f0859)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.2.0-develop-2019040519-3-g085f0859/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.2.0-develop-2019040519-3-g085f0859

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.2.0-develop-2019040519-4-geb04ba57

These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.

### job status
+ [live-roof-650-townsend runs](https://gnss-analysis.swiftnav.com/jobs/id=990603)
+ [live-roof-650-townsend-dropouts-zero-baseline runs](https://gnss-analysis.swiftnav.com/jobs/id=990604)

### gnss-analysis results
At least one run must complete for these links to have data.

#### passfail
+ [live-roof-650-townsend](https://gnss-analysis.swiftnav.com/metrics_preset=pass_fail&scenario=live-roof-650-townsend&summary_type=q50&groupby_key=firmware&firmware_versions=v2.2.0-develop-2019040519-4-geb04ba57&build_type=buildroot_pr&display_type=table)
+ [live-roof-650-townsend-dropouts-zero-baseline](https://gnss-analysis.swiftnav.com/metrics_preset=pass_fail&scenario=live-roof-650-townsend-dropouts-zero-baseline&summary_type=q50&groupby_key=firmware&firmware_versions=v2.2.0-develop-2019040519-4-geb04ba57&build_type=buildroot_pr&display_type=table)

#### detailed
+ [live-roof-650-townsend](https://gnss-analysis.swiftnav.com/metrics_preset=detailed&scenario=live-roof-650-townsend&summary_type=q50&groupby_key=firmware&firmware_versions=v2.2.0-develop-2019040519-4-geb04ba57&build_type=buildroot_pr&display_type=table)
+ [live-roof-650-townsend-dropouts-zero-baseline](https://gnss-analysis.swiftnav.com/metrics_preset=detailed&scenario=live-roof-650-townsend-dropouts-zero-baseline&summary_type=q50&groupby_key=firmware&firmware_versions=v2.2.0-develop-2019040519-4-geb04ba57&build_type=buildroot_pr&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v2.2.0-develop-2019040519-4-geb04ba57
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.2.0-develop-2019040519-4-geb04ba57)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.2.0-develop-2019040519-4-geb04ba57/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.2.0-develop-2019040519-4-geb04ba57

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.2.0-develop-2019040519-6-gf9c6da75

These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.

### job status
+ [live-roof-650-townsend runs](https://gnss-analysis.swiftnav.com/jobs/id=990847)
+ [live-roof-650-townsend-dropouts-zero-baseline runs](https://gnss-analysis.swiftnav.com/jobs/id=990848)

### gnss-analysis results
At least one run must complete for these links to have data.

#### passfail
+ [live-roof-650-townsend](https://gnss-analysis.swiftnav.com/metrics_preset=pass_fail&scenario=live-roof-650-townsend&summary_type=q50&groupby_key=firmware&firmware_versions=v2.2.0-develop-2019040519-6-gf9c6da75&build_type=buildroot_pr&display_type=table)
+ [live-roof-650-townsend-dropouts-zero-baseline](https://gnss-analysis.swiftnav.com/metrics_preset=pass_fail&scenario=live-roof-650-townsend-dropouts-zero-baseline&summary_type=q50&groupby_key=firmware&firmware_versions=v2.2.0-develop-2019040519-6-gf9c6da75&build_type=buildroot_pr&display_type=table)

#### detailed
+ [live-roof-650-townsend](https://gnss-analysis.swiftnav.com/metrics_preset=detailed&scenario=live-roof-650-townsend&summary_type=q50&groupby_key=firmware&firmware_versions=v2.2.0-develop-2019040519-6-gf9c6da75&build_type=buildroot_pr&display_type=table)
+ [live-roof-650-townsend-dropouts-zero-baseline](https://gnss-analysis.swiftnav.com/metrics_preset=detailed&scenario=live-roof-650-townsend-dropouts-zero-baseline&summary_type=q50&groupby_key=firmware&firmware_versions=v2.2.0-develop-2019040519-6-gf9c6da75&build_type=buildroot_pr&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v2.2.0-develop-2019040519-6-gf9c6da75
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.2.0-develop-2019040519-6-gf9c6da75)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.2.0-develop-2019040519-6-gf9c6da75/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.2.0-develop-2019040519-6-gf9c6da75

File: package/standalone_file_logger/standalone_file_logger/src/standalone_file_logger.cc [hunk 1]
```diff
@@ -172,45 +174,129 @@ static void save_prev_logging_fs_type_value()
...
+   *   NTFS -> FAT
+   */
+
+  fs_type_transition_t transition_type = classify_fs_transition();
+
+  switch (classify_fs_transition()) {
+  case FSTT_NO_DIFFERENCE:
+    PK_LOG_ANNO(LOG_DEBUG, "detected no difference in settings values");
+    break;
+  case FSTT_F2FS_TO_FAT: /* Fall through */
```
Comment (pmiettinen) [inline package/standalone_file_logger/standalone_file_logger/src/standalone_file_logger.cc]: Utilize `__attribute__ ((fallthrough));`?

Comment (silverjam) [reply package/standalone_file_logger/standalone_file_logger/src/standalone_file_logger.cc]: Looks like this is a GCC 7 feature 🙁

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.2.0-develop-2019040519-8-g9c0e5e46

These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.

### job status
+ [live-roof-650-townsend runs](https://gnss-analysis.swiftnav.com/jobs/id=991932)
+ [live-roof-650-townsend-dropouts-zero-baseline runs](https://gnss-analysis.swiftnav.com/jobs/id=991933)

### gnss-analysis results
At least one run must complete for these links to have data.

#### passfail
+ [live-roof-650-townsend](https://gnss-analysis.swiftnav.com/metrics_preset=pass_fail&scenario=live-roof-650-townsend&summary_type=q50&groupby_key=firmware&firmware_versions=v2.2.0-develop-2019040519-8-g9c0e5e46&build_type=buildroot_pr&display_type=table)
+ [live-roof-650-townsend-dropouts-zero-baseline](https://gnss-analysis.swiftnav.com/metrics_preset=pass_fail&scenario=live-roof-650-townsend-dropouts-zero-baseline&summary_type=q50&groupby_key=firmware&firmware_versions=v2.2.0-develop-2019040519-8-g9c0e5e46&build_type=buildroot_pr&display_type=table)

#### detailed
+ [live-roof-650-townsend](https://gnss-analysis.swiftnav.com/metrics_preset=detailed&scenario=live-roof-650-townsend&summary_type=q50&groupby_key=firmware&firmware_versions=v2.2.0-develop-2019040519-8-g9c0e5e46&build_type=buildroot_pr&display_type=table)
+ [live-roof-650-townsend-dropouts-zero-baseline](https://gnss-analysis.swiftnav.com/metrics_preset=detailed&scenario=live-roof-650-townsend-dropouts-zero-baseline&summary_type=q50&groupby_key=firmware&firmware_versions=v2.2.0-develop-2019040519-8-g9c0e5e46&build_type=buildroot_pr&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v2.2.0-develop-2019040519-8-g9c0e5e46
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.2.0-develop-2019040519-8-g9c0e5e46)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.2.0-develop-2019040519-8-g9c0e5e46/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.2.0-develop-2019040519-8-g9c0e5e46

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.2.0-develop-2019040519-12-gf6e7573b

These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.

### job status
+ [live-roof-650-townsend runs](https://gnss-analysis.swiftnav.com/jobs/id=996033)
+ [live-roof-650-townsend-dropouts-zero-baseline runs](https://gnss-analysis.swiftnav.com/jobs/id=996034)

### gnss-analysis results
At least one run must complete for these links to have data.

#### passfail
+ [live-roof-650-townsend](https://gnss-analysis.swiftnav.com/metrics_preset=pass_fail&scenario=live-roof-650-townsend&summary_type=q50&groupby_key=firmware&firmware_versions=v2.2.0-develop-2019040519-12-gf6e7573b&build_type=buildroot_pr&display_type=table)
+ [live-roof-650-townsend-dropouts-zero-baseline](https://gnss-analysis.swiftnav.com/metrics_preset=pass_fail&scenario=live-roof-650-townsend-dropouts-zero-baseline&summary_type=q50&groupby_key=firmware&firmware_versions=v2.2.0-develop-2019040519-12-gf6e7573b&build_type=buildroot_pr&display_type=table)

#### detailed
+ [live-roof-650-townsend](https://gnss-analysis.swiftnav.com/metrics_preset=detailed&scenario=live-roof-650-townsend&summary_type=q50&groupby_key=firmware&firmware_versions=v2.2.0-develop-2019040519-12-gf6e7573b&build_type=buildroot_pr&display_type=table)
+ [live-roof-650-townsend-dropouts-zero-baseline](https://gnss-analysis.swiftnav.com/metrics_preset=detailed&scenario=live-roof-650-townsend-dropouts-zero-baseline&summary_type=q50&groupby_key=firmware&firmware_versions=v2.2.0-develop-2019040519-12-gf6e7573b&build_type=buildroot_pr&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v2.2.0-develop-2019040519-12-gf6e7573b
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.2.0-develop-2019040519-12-gf6e7573b)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.2.0-develop-2019040519-12-gf6e7573b/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.2.0-develop-2019040519-12-gf6e7573b

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.2.0-develop-2019040519-15-gc19f7e66

These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.

### job status
+ [live-roof-650-townsend runs](https://gnss-analysis.swiftnav.com/jobs/id=997020)
+ [live-roof-650-townsend-dropouts-zero-baseline runs](https://gnss-analysis.swiftnav.com/jobs/id=997021)

### gnss-analysis results
At least one run must complete for these links to have data.

#### passfail
+ [live-roof-650-townsend](https://gnss-analysis.swiftnav.com/metrics_preset=pass_fail&scenario=live-roof-650-townsend&summary_type=q50&groupby_key=firmware&firmware_versions=v2.2.0-develop-2019040519-15-gc19f7e66&build_type=buildroot_pr&display_type=table)
+ [live-roof-650-townsend-dropouts-zero-baseline](https://gnss-analysis.swiftnav.com/metrics_preset=pass_fail&scenario=live-roof-650-townsend-dropouts-zero-baseline&summary_type=q50&groupby_key=firmware&firmware_versions=v2.2.0-develop-2019040519-15-gc19f7e66&build_type=buildroot_pr&display_type=table)

#### detailed
+ [live-roof-650-townsend](https://gnss-analysis.swiftnav.com/metrics_preset=detailed&scenario=live-roof-650-townsend&summary_type=q50&groupby_key=firmware&firmware_versions=v2.2.0-develop-2019040519-15-gc19f7e66&build_type=buildroot_pr&display_type=table)
+ [live-roof-650-townsend-dropouts-zero-baseline](https://gnss-analysis.swiftnav.com/metrics_preset=detailed&scenario=live-roof-650-townsend-dropouts-zero-baseline&summary_type=q50&groupby_key=firmware&firmware_versions=v2.2.0-develop-2019040519-15-gc19f7e66&build_type=buildroot_pr&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v2.2.0-develop-2019040519-15-gc19f7e66
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.2.0-develop-2019040519-15-gc19f7e66)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.2.0-develop-2019040519-15-gc19f7e66/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.2.0-develop-2019040519-15-gc19f7e66

**Comment by silverjam** [comment]
Obsoleted in favor of https://github.com/swift-nav/piksi_buildroot/pull/1231

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.2.0-develop-2019040519-16-g19922a5f

These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.

### job status
+ [live-roof-650-townsend runs](https://gnss-analysis.swiftnav.com/jobs/id=999931)
+ [live-roof-650-townsend-dropouts-zero-baseline runs](https://gnss-analysis.swiftnav.com/jobs/id=999932)

### gnss-analysis results
At least one run must complete for these links to have data.

#### passfail
+ [live-roof-650-townsend](https://gnss-analysis.swiftnav.com/metrics_preset=pass_fail&scenario=live-roof-650-townsend&summary_type=q50&groupby_key=firmware&firmware_versions=v2.2.0-develop-2019040519-16-g19922a5f&build_type=buildroot_pr&display_type=table)
+ [live-roof-650-townsend-dropouts-zero-baseline](https://gnss-analysis.swiftnav.com/metrics_preset=pass_fail&scenario=live-roof-650-townsend-dropouts-zero-baseline&summary_type=q50&groupby_key=firmware&firmware_versions=v2.2.0-develop-2019040519-16-g19922a5f&build_type=buildroot_pr&display_type=table)

#### detailed
+ [live-roof-650-townsend](https://gnss-analysis.swiftnav.com/metrics_preset=detailed&scenario=live-roof-650-townsend&summary_type=q50&groupby_key=firmware&firmware_versions=v2.2.0-develop-2019040519-16-g19922a5f&build_type=buildroot_pr&display_type=table)
+ [live-roof-650-townsend-dropouts-zero-baseline](https://gnss-analysis.swiftnav.com/metrics_preset=detailed&scenario=live-roof-650-townsend-dropouts-zero-baseline&summary_type=q50&groupby_key=firmware&firmware_versions=v2.2.0-develop-2019040519-16-g19922a5f&build_type=buildroot_pr&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v2.2.0-develop-2019040519-16-g19922a5f
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.2.0-develop-2019040519-16-g19922a5f)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.2.0-develop-2019040519-16-g19922a5f/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.2.0-develop-2019040519-16-g19922a5f

**Pull request closed without merging**